src/librustc/ty/layout.rs

```diff
@@ -697,6 +697,7 @@ impl<'a, 'tcx> LayoutCx<'tcx, TyCtxt<'a, 'tcx, 'tcx>> {
                     }
 
                     let mut size = Size::ZERO;
+                    let mut abi = Abi::Uninhabited;
```

Comment:
What is the meaining of this initial value?

Notice that a union should NOT be uninhabited even if all of its fields are! That would also be a form of "propagating ranges".

Reply:
Ah, you are using it as a sentinel value for "nothing found yet". This warrants a comment. However, we should also *never* use the `Uninhabited` layout on unions. So maybe better use an `Option` here, initialized to `None`?